runtime/drive_handler.go

```diff
@@ -125,10 +125,20 @@ func (h *stubDriveHandler) createStubDrive(driveID, path string) error {
 	return nil
 }
 
+func filterOnlyStubDrives(drives []models.Drive) []models.Drive {
+	stubDrives := make([]models.Drive, 0)
+	for _, d := range drives {
+		if !*d.IsRootDevice {
+			stubDrives = append(stubDrives, d)
+		}
+	}
+	return stubDrives
+}
+
 // SetDrives will set the given drives and the offset to which the stub drives
 // start.
 func (h *stubDriveHandler) SetDrives(offset int64, d []models.Drive) {
-	h.drives = d
+	h.drives = filterOnlyStubDrives(d)
```

Comment:
What's the error now when you encounter the issue in #230 ?

The fix here is good either way and if the error is still unclear that can be addressed separately, just curious if it's changed to something more comprehensible.

Reply:
Good point. Let me try that.

Reply:
Now the error mentions "failed to patch stub drive: There are no remaining drives to be used" which is better than what I had before

```
--- FAIL: TestCreateContainerWithSameName_Isolated (33.06s)
    require.go:794:
                Error Trace:    service_integ_test.go:565
                                                        service_integ_test.go:655
                                                        service_integ_test.go:676
                Error:          Received unexpected error:
                                unknown
                                github.com/containerd/containerd/errdefs.init.ializers
                                        /go/pkg/mod/github.com/containerd/containerd@v1.2.1-0.20190711201753-f2b6c31d0fa7/errdefs/errors.go:43
                                runtime.main
                                        /usr/local/go/src/runtime/proc.go:188
                                runtime.goexit
                                        /usr/local/go/src/runtime/asm_amd64.s:1337
                                failed to patch stub drive: There are no remaining drives to be used
                                github.com/containerd/containerd/errdefs.FromGRPC
                                        /go/pkg/mod/github.com/containerd/containerd@v1.2.1-0.20190711201753-f2b6c31d0fa7/errdefs/grpc.go:107
                                github.com/containerd/containerd.(*container).NewTask
                                        /go/pkg/mod/github.com/containerd/containerd@v1.2.1-0.20190711201753-f2b6c31d0fa7/container.go:286
                                github.com/firecracker-microvm/firecracker-containerd/runtime.startAndWaitTask
                                        /firecracker-containerd/runtime/service_integ_test.go:564
                                github.com/firecracker-microvm/firecracker-containerd/runtime.testCreateContainerWithSameName
                                        /firecracker-containerd/runtime/service_integ_test.go:655
                                github.com/firecracker-microvm/firecracker-containerd/runtime.TestCreateContainerWithSameName_Isolated
                                        /firecracker-containerd/runtime/service_integ_test.go:676
                                testing.tRunner
                                        /usr/local/go/src/testing/testing.go:865
                                runtime.goexit
                                        /usr/local/go/src/runtime/asm_amd64.s:1337
                Test:           TestCreateContainerWithSameName_Isolated
                Messages:       failed to create task for container TestCreateContainerWithSameName_Isolated-1565126341850109350
FAIL
exit status 1
```

Reply:
Nice, yeah that's much clearer than before